core/node/libp2p/rcmgr_defaults.go

```diff
@@ -79,109 +49,84 @@ Run 'ipfs swarm limit all' to see the resulting limits.
-		ServiceBaseLimit:     infiniteBaseLimit,
-		ServiceLimitIncrease: noLimitIncrease,
+		ServiceDefault: infiniteResrouceLimits,
 
-		ServicePeerBaseLimit:     infiniteBaseLimit,
-		ServicePeerLimitIncrease: noLimitIncrease,
+		ServicePeerDefault: infiniteResrouceLimits,
 
-		ProtocolBaseLimit:     infiniteBaseLimit,
-		ProtocolLimitIncrease: noLimitIncrease,
+		ProtocolDefault: infiniteResrouceLimits,
 
-		ProtocolPeerBaseLimit:     infiniteBaseLimit,
-		ProtocolPeerLimitIncrease: noLimitIncrease,
+		ProtocolPeerDefault: infiniteResrouceLimits,
 
-		ConnBaseLimit:     infiniteBaseLimit,
-		ConnLimitIncrease: noLimitIncrease,
+		Conn: infiniteResrouceLimits,
 
-		StreamBaseLimit:     infiniteBaseLimit,
-		StreamLimitIncrease: noLimitIncrease,
+		Stream: infiniteResrouceLimits,
 
 		// Limit the resources consumed by a peer.
 		// This doesn't protect us against intentional DoS attacks since an attacker can easily spin up multiple peers.
 		// We specify this limit against unintentional DoS attacks (e.g., a peer has a bug and is sending too much traffic intentionally).
 		// In that case we want to keep that peer's resource consumption contained.
 		// To keep this simple, we only constrain inbound connections and streams.
-		PeerBaseLimit: rcmgr.BaseLimit{
-			Memory:          bigEnough,
-			FD:              bigEnough,
-			Conns:           bigEnough,
-			ConnsInbound:    rcmgr.DefaultLimits.PeerBaseLimit.ConnsInbound,
-			ConnsOutbound:   bigEnough,
-			Streams:         bigEnough,
-			StreamsInbound:  rcmgr.DefaultLimits.PeerBaseLimit.StreamsInbound,
-			StreamsOutbound: bigEnough,
-		},
-		// Most limits don't see an increase because they're already infinite/bigEnough.
-		// The values that should scale based on the amount of memory allocated to libp2p need to increase accordingly.
-		PeerLimitIncrease: rcmgr.BaseLimitIncrease{
-			Memory:          0,
-			FDFraction:      0,
-			Conns:           0,
-			ConnsInbound:    rcmgr.DefaultLimits.PeerLimitIncrease.ConnsInbound,
-			ConnsOutbound:   0,
-			Streams:         0,
-			StreamsInbound:  rcmgr.DefaultLimits.PeerLimitIncrease.StreamsInbound,
-			StreamsOutbound: 0,
+		PeerDefault: rcmgr.ResourceLimits{
+			Memory:          rcmgr.Unlimited64,
+			FD:              rcmgr.Unlimited,
+			Conns:           rcmgr.Unlimited,
+			ConnsInbound:    rcmgr.DefaultLimit,
+			ConnsOutbound:   rcmgr.Unlimited,
+			Streams:         rcmgr.Unlimited,
+			StreamsInbound:  rcmgr.DefaultLimit,
+			StreamsOutbound: rcmgr.Unlimited,
 		},
 	}
 
-	// Whatever limits libp2p has specifically tuned for its protocols/services we'll apply.
+	scalingLimitConfig := rcmgr.DefaultLimits
 	libp2p.SetDefaultServiceLimits(&scalingLimitConfig)
 
-	orig := scalingLimitConfig.Scale(int64(maxMemory), maxFD)
-	defaultLimitConfig := orig.ToPartialLimitConfig()
+	partialLimits = partialLimits.Build(scalingLimitConfig.Scale(int64(maxMemory), maxFD)).ToPartialLimitConfig()
```

Comment:
I think a clarifying comment here could help...  This is my understanding of what is happening (which I agree is what we want, but I think it could help to clearly state our intent).

```suggestion
	// Anything set above in partialLimits that had a value of rcmgr.DefaultLimit will be overridden.
	// Anything in scalingLimitConfig that wasn't defined in partialLimits above will be added (e.g., libp2p's default service limits).
	partialLimits = partialLimits.Build(scalingLimitConfig.Scale(int64(maxMemory), maxFD)).ToPartialLimitConfig()
```